src/neo/SmartContract/NefFile.cs

```diff
@@ -80,7 +86,7 @@ private void SerializeHeader(BinaryWriter writer)
             writer.Write(Version.Build);
             writer.Write(Version.Revision);
 
-            writer.Write(ScriptHash);
+            writer.WriteVarString(Abi.ToJson().ToString());
```

Comment:
Why are serializing the ABI as a json string inside the binary NEF file? Wouldn't it be more efficient to serialize this information out in binary format?

Reply:
Like .net file format. https://www.codeproject.com/Articles/12585/The-NET-File-Format#MetaTables